lsp-mode.el

```diff
@@ -5646,29 +5646,32 @@ perform the request synchronously."
            (lsp-request "workspace/symbol" `(:query ,pattern))))
 
 (defun lsp--get-symbol-to-rename ()
-  "Get symbol to rename and placeholder at point."
-  (if (let ((rename-provider (or (lsp--capability :renameProvider)
-                                 (-some-> (lsp--registered-capability "textDocument/rename")
-                                   (lsp--registered-capability-options)))))
-        (lsp:rename-options-prepare-provider? rename-provider))
-      (-when-let (response (lsp-request "textDocument/prepareRename"
-                                        (lsp--text-document-position-params)))
+  "Get symbol to rename and placeholder at point.
+Returns a cons (SYMBOL-TO-RENAME . PLACEHOLDER). Returns nil if
+renaming is supported but cannot be done at point."
+  (unless (lsp-feature? "textDocument/rename")
+    (error "The connected server(s) doesn't support renaming"))
+  (if (lsp-feature? "textDocument/prepareRename")
+      (when-let ((response
+                  (lsp-request "textDocument/prepareRename"
+                               (lsp--text-document-position-params))))
         (-let* (((start . end) (lsp--range-to-region
                                 (if (lsp-range? response)
                                     response
                                   (lsp:prepare-rename-result-range response))))
-                (symbol (buffer-substring-no-properties start end))
+                (symbol (buffer-substring start end))
```

Comment:
Why we need to use `buffer-substring` instead of `-no-properties` version?
Do we need the highlighting when prompting?

Reply:
Compare:
![20201110-075543](https://user-images.githubusercontent.com/69264154/98640368-5f55f700-232a-11eb-8117-1e98e030e001.png)
VS with highlighting:
![20201110-075458](https://user-images.githubusercontent.com/69264154/98640466-654bd800-232a-11eb-96dd-c80f54075d02.png)

Do you prefer the former?